kornia/augmentation/augmentation.py

```diff
@@ -586,21 +675,20 @@ def __repr__(self) -> str:
         return self.__class__.__name__ + f"({repr}, {super().__repr__()})"
 
     def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, torch.Tensor]:
-        if isinstance(self.size, tuple):
-            size_param = (self.size[0], self.size[1])
-        elif isinstance(self.size, int):
-            size_param = (self.size, self.size)
-        else:
-            raise Exception(f"Invalid size type. Expected (int, tuple(int, int). "
-                            f"Got: {type(self.size)}.")
         return rg.center_crop_generator(
-            batch_shape[0], batch_shape[-2], batch_shape[-1], size_param, self.device)
+            batch_shape[0], batch_shape[-2], batch_shape[-1], self.size, self.device)
 
     def compute_transformation(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
-        return F.compute_crop_transformation(input, params, self.flags)
+        transform: torch.Tensor = get_perspective_transform(
+            params['src'].to(input), params['dst'].to(input))
+        transform = transform.expand(input.shape[0], -1, -1)
+        return transform
 
-    def apply_transform(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
-        return F.apply_crop(input, params, self.flags)
+    def apply_transform(
+        self, input: torch.Tensor, params: Dict[str, torch.Tensor], transform: Optional[torch.Tensor] = None
+    ) -> torch.Tensor:
+        return crop_by_transform_mat(
```

Comment:
Why don’t use directly warp_affine ?

Reply:
If so, there gonna be many code duplications for all the cropping and perspective.

Reply:
Then add tests for that function